src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateSearchDao.java

```diff
@@ -7,159 +7,98 @@
     }
 
     public List<Mandate> search(MandateSearchParams mandateSearchParams) {
-        Params params = new Params(mandateSearchParams);
-        String searchExtraFields = generateQuery(params);
+        var sql = new StringBuilder(2048);
+        sql.append("SELECT DISTINCT" +
+                "  m.id AS mandate_id," +
+                "  m.external_id AS mandate_external_id," +
+                "  m.mandate_reference AS mandate_mandate_reference," +
+                "  m.service_reference AS mandate_service_reference," +
+                "  m.gateway_account_id AS mandate_gateway_account_id," +
+                "  m.return_url AS mandate_return_url," +
+                "  m.state AS mandate_state," +
+                "  m.created_date AS mandate_created_date," +
+                "  m.payment_provider_id AS mandate_payment_provider_id," +
+                "  g.id AS gateway_account_id," +
+                "  g.external_id AS gateway_account_external_id," +
+                "  g.payment_provider AS gateway_account_payment_provider," +
+                "  g.type AS gateway_account_type," +
+                "  g.description AS gateway_account_description," +
+                "  g.analytics_id AS gateway_account_analytics_id," +
+                "  g.access_token AS gateway_account_access_token," +
+                "  g.organisation AS gateway_account_organisation," +
+                "  p.id AS payer_id," +
+                "  p.mandate_id AS payer_mandate_id," +
+                "  p.external_id AS payer_external_id," +
+                "  p.name AS payer_name," +
+                "  p.email AS payer_email," +
+                "  p.bank_account_number_last_two_digits AS payer_bank_account_number_last_two_digits," +
+                "  p.bank_account_requires_authorisation AS payer_bank_account_requires_authorisation," +
+                "  p.bank_account_number AS payer_bank_account_number," +
+                "  p.bank_account_sort_code AS payer_bank_account_sort_code," +
+                "  p.bank_name AS payer_bank_name," +
+                "  p.created_date AS payer_created_date" +
+                " FROM mandates m" +
+                "  JOIN gateway_accounts g ON g.id = m.gateway_account_id " +
+                "  LEFT JOIN payers p ON p.mandate_id = m.id " +
+                "WHERE g.external_id = :gatewayAccountExternalId");
+        //            "  ORDER BY p.id DESC OFFSET :offset LIMIT :limit";
+
+        var sqlParams = new HashMap<String, Object>();
+        sqlParams.put("gatewayAccountExternalId", mandateSearchParams.getGatewayAccountExternalId());
+
+        mandateSearchParams.getReference().filter(s -> !s.isBlank()).ifPresent(serviceReference -> {
+            sql.append(" AND m.service_reference ILIKE :serviceReference");
+            sqlParams.put("serviceReference", "%" + serviceReference + "%");
+        });
+        
+        mandateSearchParams.getMandateBankStatementReference().ifPresent(bankStatementRef -> {
```

Comment:
Does this want an `.filter(s -> !s.isBlank())` too?

Reply:
fixed